File: app/src/main/java/com/uni/phamduy/massagefinder/module/place/Place.java

package com.uni.phamduy.massagefinder.module.place;

import java.util.List;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Place {

    @SerializedName("id")
    @Expose
    private String id;
    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("address")
    @Expose
    private Address address;
    @SerializedName("review")
    @Expose
    private String review;
    @SerializedName("rating")
    @Expose
    private Integer rating;
    @SerializedName("tickets")
    @Expose
    private List<Ticket> tickets = null;
    @SerializedName("hotLines")
    @Expose
    private List<String> hotLines = null;
    @SerializedName("tags")
    @Expose
    private List<Object> tags = null;
    @SerializedName("coverImage")
    @Expose
    private CoverImage coverImage;
    @SerializedName("slideImages")
    @Expose
    private List<SlideImage> slideImages = null;
    @SerializedName("distance")
    @Expose
    private String distance;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Address getAddress() {
        return address;
    }

    public void setAddress(Address address) {
        this.address = address;
    }

    public String getReview() {
        return review;
    }

    public void setReview(String review) {
        this.review = review;
    }

    public Integer getRating() {
        return rating;
    }

    public void setRating(Integer rating) {
        this.rating = rating;
    }

    public List<Ticket> getTickets() {
        return tickets;
    }

    public void setTickets(List<Ticket> tickets) {
        this.tickets = tickets;
    }

    public List<String> getHotLines() {
        return hotLines;
    }

    public void setHotLines(List<String> hotLines) {
        this.hotLines = hotLines;
    }

    public List<Object> getTags() {
        return tags;
    }

    public void setTags(List<Object> tags) {
        this.tags = tags;
    }

    public CoverImage getCoverImage() {
        return coverImage;
    }

    public void setCoverImage(CoverImage coverImage) {
        this.coverImage = coverImage;
    }

    public List<SlideImage> getSlideImages() {
        return slideImages;
    }

    public void setSlideImages(List<SlideImage> slideImages) {
        this.slideImages = slideImages;
    }

    public String getDistance() {
        return distance;
    }

    public void setDistance(String distance) {
        this.distance = distance;
    }

}

File: app/src/main/java/com/uni/phamduy/massagefinder/module/place/Place.kt
package com.uni.phamduy.massagefinder.module.place

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Place() : Parcelable {

    @SerializedName("id")
    @Expose
    var id: String? = null
    @SerializedName("name")
    @Expose
    var name: String? = null
    @SerializedName("address")
    @Expose
    var address: Address? = null
    @SerializedName("review")
    @Expose
    var review: String? = null
    @SerializedName("rating")
    @Expose
    var rating: Int? = null
    @SerializedName("tickets")
    @Expose
    var tickets: List<Ticket>? = null
    @SerializedName("hotLines")
    @Expose
    var hotLines: List<String>? = null
    @SerializedName("tags")
    @Expose
    var tags: List<Any>? = null
    @SerializedName("coverImage")
    @Expose
    var coverImage: CoverImage? = null
    @SerializedName("slideImages")
    @Expose
    var slideImages: List<SlideImage>? = null
    @SerializedName("distance")
    @Expose
    var distance: String? = null

    constructor(parcel: Parcel) : this() {
        id = parcel.readString()
        name = parcel.readString()
        address = parcel.readParcelable(Address::class.java.classLoader)
        review = parcel.readString()
        rating = parcel.readValue(Int::class.java.classLoader) as? Int
        tickets = parcel.createTypedArrayList(Ticket)
        hotLines = parcel.createStringArrayList()
        coverImage = parcel.readParcelable(CoverImage::class.java.classLoader)
        slideImages = parcel.createTypedArrayList(SlideImage)
        distance = parcel.readString()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(id)
        parcel.writeString(name)
        parcel.writeParcelable(address, flags)
        parcel.writeString(review)
        parcel.writeValue(rating)
        parcel.writeTypedList(tickets)
        parcel.writeStringList(hotLines)
        parcel.writeParcelable(coverImage, flags)
        parcel.writeTypedList(slideImages)
        parcel.writeString(distance)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Place> {
        override fun createFromParcel(parcel: Parcel): Place {
            return Place(parcel)
        }

        override fun newArray(size: Int): Array<Place?> {
            return arrayOfNulls(size)
        }
    }


}
